lib/measured/conversion_table_builder.rb

```diff
@@ -1,14 +1,29 @@
 class Measured::ConversionTableBuilder
   attr_reader :units
 
-  def initialize(units)
+  def initialize(units, cache: nil)
     @units = units
+    cache ||= { class: Measured::Cache::Null }
+    @cache = cache[:class].new(*cache[:args])
   end
 
   def to_h
-    table = {}
+    return @cache.read if @cache.exist?
+    generate_table
+  end
+
+  def update_cache
+    @cache.write(generate_table)
+  end
+
+  def cached?
+    @cache.exist?
+  end
 
-    units.map{|u| u.name}.each do |to_unit|
+  private
+
+  def generate_table
+    units.map{|u| u.name}.each_with_object({}) do |to_unit, table|
```

Comment:
nit: just to be consistent now that we are changing it, what do you think about doing `units.map(&:name)` instead?